include/deal.II/particles/generators.h

```diff
@@ -63,6 +72,99 @@ namespace Particles
       const Mapping<dim, spacedim> &      mapping =
         StaticMappingQ1<dim, spacedim>::mapping);
 
+    /**
+     * A function that generates one particle at a random location in cell @p cell and with
+     * index @p id. The function expects a random number generator to avoid the expensive generation
+     * and destruction of a generator for every particle and optionally takes
+     * into account a mapping for the cell. The algorithm implemented in the
+     * function is described in
+     *
+     * Gassmoeller, R., Lokavarapu, H., Heien, E., Puckett, E. G., & Bangerth,
+     * W. (2018). Flexible and Scalable Particle‐in‐Cell Methods With Adaptive
+     * Mesh Refinement for Geodynamic Computations. Geochemistry, Geophysics,
+     * Geosystems, 19(9), 3596-3604.
+     *
+     * @param cell The cell in which a particle is generated.
+     *
+     * @param id The particle index that will be assigned to the new particle.
+     *
+     * @param random_number_generator A random number generator that will be
+     * used for the creation of th particle.
+     *
+     * @param mapping An optional mapping object that is used to map reference
+     * location in the unit cell to the real cell. If no
+     * mapping is provided a MappingQ1 is assumed.
+     */
+    template <int dim, int spacedim = dim>
+    Particle<dim, spacedim>
+    random_particle_in_cell(
+      const typename Triangulation<dim, spacedim>::active_cell_iterator &cell,
+      const types::particle_index                                        id,
+      boost::mt19937 &              random_number_generator,
```

Comment:
We can now use the C++11 class `std::mt19937`. You'll have to `#include` the file `<random>` but in return get rid of the three lines for the BOOST include.